[Cleanup]adds handling for messages cont. emojis/links

kablekompany opened the pull request:
### Type

- [ ] Bugfix
- [ ] Enhancement
- [x] New feature

### Description of the changes
`[p]cleanup emoji`
`[p]cleanup links`

each default to searching for 10 messages respectively

Blocked by #4728 (`[p]cleanup stickers`)

kablekompany committed:
[Cleanup]handle messages containing emojis/links

kablekompany committed:
Update cleanup.py

kablekompany committed:
Update cleanup.py

kablekompany committed:
can't imagine forgetting import

kablekompany committed:
Update cleanup.py

kablekompany commented:
Have made this change on my own red, and can confirm no false positives as of yet on link or emoji clearing (unicode emojis of course are not purged). Initially I had an alias with `cleanup text ":" to sort of target either but couldn't settle for one or the other, and it would always clean both. This has been tested in a moderation environment where chat speed is variable of 1-2msgs per second with enforced slowmode of 30s (so you can imagine the spam, ads, emoji overuse encountered). 

Just wanted to validate positive test to date and provide operational stroke of proposed addition, regardless if there is intent or desire for use. 

Thanks!

kablekompany committed:
Update cleanup.py

kablekompany committed:
bring up to date

kablekompany committed:
Update cleanup.py

kablekompany committed:
reversion of v3/develop

Jackenmen commented on redbot/cogs/cleanup/cleanup.py:
```diff
@@ -10,12 +10,18 @@
 
 log = logging.getLogger("red.cleanup")
 
+LINKS = re.compile(r"http[s]?://(?:[a-zA-Z]|[0-9]|[$-_@.&+]|[!*(),]|(?:%[0-9a-fA-F][0-9a-fA-F]))+")
```
We have a regex for links in `redbot.core.utils.common_filters.URL_RE` which could be used here.

Jackenmen commented on redbot/cogs/cleanup/cleanup.py:
```diff
@@ -115,13 +121,13 @@ def message_filter(message):
         return collected
 
     @commands.group()
-    @checks.mod_or_permissions(manage_messages=True)
```
Could you merge current `V3/develop` into this branch so that these differences don't show up in this diff?

Jackenmen reviewed (commented):
This is not a review, I just had two things I wanted to mention. Feel free to ignore this until actual review comes up.

kablekompany committed:
request

kablekompany commented:
applied the noted changes (i think) 
let me know if that diff is fucky (git is confusing for me at times 😓 )

kablekompany commented:
moved emoji regex to core and added unicode emoji regex extracting to core as well
Can revert if these properties are not sufficient for core otherwise. 

making the unicode regex accessible could open up use cases in automod filters or further expanding cleanup functions.

Drapersniper commented on redbot/cogs/cleanup/cleanup.py:
```diff
@@ -517,7 +512,6 @@ def check(m):
         await mass_purge(to_delete, channel)
 
     @cleanup.command(name="self")
-    @check_self_permissions()
```
I dont think this was intentional, this is the reason that `    @checks.mod_or_permissions(manage_messages=True)` isnt added to the group like you done in this PR. I dont know enough about this check to comment on why just something i saw when skimming through it.

Jackenmen replied:
Looks to me like a merge that went south. The checks in cleanup cog were touched somewhat recently and this seems to be accidentally reverting it.

kablekompany replied:
'Twas a revert to remove the diff from PR #4481 from @Predeactor as git was not attributing the correct blame. Thought reverting the commit of adding those and merging in would fix but ig not...
oh boi guise i'm unsure how to fix without closing and reopening with a with fresh V3/develop base 😟

kablekompany replied:
Let me know if that's best course, and I'll dip this PR, burn the branch, base from current V3/develop, write in the changes, and open new PR.

Sorry! ha

kablekompany committed:
let's try and fix this fuckery

kablekompany committed:
okay — i sweatergod

kablekompany committed:
lol

kablekompany committed:
blue on black, tears on a river, push on a shove, it don't mean much

kablekompany commented:
successfully got the git blame back on track

kablekompany committed:
docstrings, import correction

kablekompany committed:
yeet

kablekompany committed:
kinda necessary lol

kablekompany committed:
yeet the re, and ready for review

kablekompany committed:
Merge remote-tracking branch 'redbot/V3/develop' into patch-1

* redbot/V3/develop:
  Automated Crowdin downstream (#4567)
  Fix leaderboard error in DMs for global banks (#4569)

kablekompany committed:
Merge remote-tracking branch 'redbot/V3/develop' into patch-1

* redbot/V3/develop:
  Fix `reason` variable shadowing in `[p]massban` (#4575)
  Automated Crowdin downstream (#4578)

kablekompany committed:
Merge remote-tracking branch 'redbot/V3/develop' into patch-1

* redbot/V3/develop:
  Bump versions of actions used by our workflows (#4586)
  [Docs] CogManagerUI user guide (#4152)

kablekompany committed:
Merge branch origin/V3/develop into patch-1

kablekompany committed:
Merge branch origin/V3/develop into patch-1

kablekompany committed:
Merge branch origin/V3/develop into patch-1

kablekompany committed:
adds support for `[p]cleanup stickers`

kablekompany committed:
Merge branch upstream/V3/develop into patch-1

kablekompany commented:
Added support for cleanup stickers 

Cause those were a fantastic idea and definitely not spammed :XtoDoubt:

kablekompany committed:
Merge branch origin/V3/develop into patch-1

kablekompany commented:
Will discontinue pulling in base unless otherwise expressed this is a potential wanted merge. Just to save us all some email notifs 😅 

will rectify a final updated upstream when/if that review is requested. Cheers!

kablekompany committed:
Merge branch upstream/V3/develop into patch-1
